### ISSUE TITLE
Question about the gravity loss.

### ISSUE THREAD

**Issue by liren2515** [isantesteban/snug#4]
Hello Igor,

Thanks for your sharing of the codes for the physics-based losses. This work is really amazing.
I checked your code and found that when you load the motion data, you first rotate the root pose. I guess this rotation is used to make sure you have the desired `x-y-z` coordinate.
```
swap_rotation = R.from_euler('zx', [-90, 270], degrees=True)
root_rot = R.from_rotvec(pose[:, :3])
pose[:, :3] = (swap_rotation * root_rot).as_rotvec()
trans = swap_rotation.apply(trans)
```

And in your `gravitational_energy`, which computes the loss for the gravity, you use the values of `x[:, :, 1]`. Since the last dimension of x represents the `[x,y,z]` position, this means `y` should be the height of the vertices. Do I understand this correctly?
```
def gravitational_energy(x, mass, g=9.81, return_average=True):
    batch_size = tf.cast(tf.shape(x)[0], x.dtype)
    U = g * mass[tf.newaxis, tf.newaxis] * x[:, :, 1]

    if return_average:
        return tf.reduce_sum(U) / batch_size

    return tf.reduce_sum(U, axis=-1)
```

However, when I show the body mesh before and after the rotation with Blender, I get the following results.

Before rotation:
![before](https://user-images.githubusercontent.com/23461830/166693856-4ab4e0fb-781e-4386-a452-65ee50c54597.jpg)

After rotation:
![after](https://user-images.githubusercontent.com/23461830/166694048-933d2324-6558-4f81-b3f2-47d5901c062f.jpg)

It looks like after rotation, x-axis represents the height of vertices instead of y-axis. Did I misunderstand something here?

**Comment by isantesteban** [comment]
Hi! Thanks for your kind words

In our work we use the y-axis as the height, but the motions from CMU usually use the z-axis, so that's the reason why we apply a rotation to the root. The pose that you are using seems to be already in the correct axes so you may need to comment the rotation code to get the correct results. Out of curiosity, is that sequence from CMU or is it from another motion capture database? 

Also, Blender uses the z axis as 'up' so depending on how you import the meshes you may see them in the wrong direction. This is how I load them:
![image](https://user-images.githubusercontent.com/17111347/166703299-872620d2-7dfc-40c8-ad94-7135ecb218c6.png)

And this is what I see if I load the first frame of sequence 01_01 from CMU:
![image](https://user-images.githubusercontent.com/17111347/166705761-9fe441a6-4842-4234-a5fd-9c78c623f838.png)

**Comment by liren2515** [comment]
> Also, Blender uses the z axis as 'up' so depending on how you import the meshes you may see them in the wrong direction. This is how I load them:

I just realized this. Thanks for pointing it out. 

The pose I loaded is from CMU dataset.  So when you say 'the motions from CMU usually use the z-axis', does that means there are some sequences in CMU that don't use the z-axis and I have to check them one by one?

**Comment by dancasas** [comment]
I think al CMU sequences converted to SMPL in AMASS should use the same axis convention.

**Comment by liren2515** [comment]
> I think al CMU sequences converted to SMPL in AMASS should use the same axis convention.

Thanks. I got it.
